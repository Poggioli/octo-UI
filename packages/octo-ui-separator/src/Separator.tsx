import { forwardRef } from "@octo-ui/core";
import { StyledSeparatorRoot } from "./styles";
import { SeparatorProps } from "./types";

const Separator = forwardRef<typeof StyledSeparatorRoot, SeparatorProps>(
  ({ color, ...props }, forwardedRef) => {
    const css = {
      backgroundColor: color ?? "$grey6",
      ...props.css,
    };

    return <StyledSeparatorRoot {...props} ref={forwardedRef} css={css} />;
  }
);
Separator.toString = () => `.${StyledSeparatorRoot.className}`;

export { StyledSeparatorRoot };
export { Separator };

