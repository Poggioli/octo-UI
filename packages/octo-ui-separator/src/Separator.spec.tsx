import { render } from "@testing-library/react";
import { Separator } from "./Separator";

describe("<Separator />", () => {
  it(`GIVEN a <Separator />
      WHEN render is called
      THEN should render a <Separator />`, () => {
    const rendered = render(<Separator />);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Separator />
      WHEN render is called
      THEN should render a <Separator />`, () => {
    const rendered = render(<Separator color="$colors$primary9" />);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Separator />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Separator.toString()).toBe(".octo-UI-c-hxgBlx");
  });
});
