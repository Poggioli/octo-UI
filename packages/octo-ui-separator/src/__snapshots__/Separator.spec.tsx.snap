// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> GIVEN a <Separator />
      WHEN render is called
      THEN should render a <Separator /> 1`] = `
<div>
  <div
    class="octo-UI-c-hxgBlx octo-UI-c-hxgBlx-iiyWNbn-css"
    data-orientation="horizontal"
    role="separator"
  />
</div>
`;

exports[`<Separator /> GIVEN a <Separator />
      WHEN render is called
      THEN should render a <Separator /> 2`] = `
<div>
  <div
    class="octo-UI-c-hxgBlx octo-UI-c-hxgBlx-ifhNQnn-css"
    data-orientation="horizontal"
    role="separator"
  />
</div>
`;
